### PR TITLE
feat: add-option-to-auto-start-mining-after-app-startup

### PR DESCRIPTION
**Pull request by Misieq01**
### [ Features ]
- Added `mine_on_app_start` flag for automatically start mining after app setup finish
- Added option in settings for toggling mentioned flag

- Added `should_auto_launch` flag which either enable or disable auto starting tari app with system boot
- Added option in settings for toggling mentioned flag 
------------
#### `mine_on_app_start`
- Enable or disable option in settings and restart app
------------
#### `should_auto_launch`
- If testing on production build that includes this change just enable option in settings and restart system
- If testing on dev then:
- ##### MacOS
   - Run `npm run tauri build`
   - Go to src-tauri/target/release/bundle/dmg/Tari Universe_[version]_aarch64.dmg
   - Install app in applications
   - Run and do first step
- ##### Windows
   - Run `npm run tauri build`
   - Uninstall tari app
   - Go to src-tauri/target/release/bundle/msi/Tari Universe_[version]_x64.msi
   - Install app
   - Run and do first step
- ##### Linux
   - Run `npm run tauri build`
   - Run `sudo apt-get remove tari-universe`
   - Go to src-tauri/target/release/bundle/deb
   - Run in terminal at location `sudo dpkg -i tari-universe_[version]_amd64.deb`
   - Run and do first step
------------
### [ Issues ]
- On Windows sometime node_manager cannot start throwing:
`ERROR Could not start node manager: UnknownError(Process watcher task has already finished) // src/main.rs:564`
- On MacOS when auto_launch option is enabled and we restart with option to resume all applications then we sometimes can end up with two instances of our application running. It's probably fixed but need more feedback

### CODE REVIEW

File: src/containers/SideBar/components/Settings/MiningSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export default function MiningSettings() {
+    const { t } = useTranslation(['settings'], { useSuspense: false });
+
+    const { mine_on_app_start } = useAppStatusStore((s) => ({
+        mine_on_app_start: s.mine_on_app_start,
+    }));
+
+    const toggleMiningOnAppStart = (event: ChangeEvent<HTMLInputElement>) => {
+        console.log('Toggling mining on app start', event.target.checked);
```
Comment (Krakaw) [inline src/containers/SideBar/components/Settings/MiningSettings.tsx:17]: We can remove this
```suggestion
```

**Review by Krakaw — APPROVED** [review]
LGTM

**Comment by Krakaw** [comment]
@Misieq01 please can you fix the conflicts and push.

**Comment by stringhandler** [comment]
This is not 100% solving the issue #122. When I restart my computer, Tari Universe must also restart. The user should be given an option in settings:
* [ ] Start Tari Universe when I log on

Which when ticked, should start when a user reboots or logs out of windows and logs in again. 
We only need to target Macos and Windows for this.

**Review by stringhandler — CHANGES_REQUESTED** [review]
See comment

**Review by brianp — APPROVED** [review]
Tested on MacOS.

Validated mine on start setting did not mine on start when toggled off, and did mine on start when toggled on.

Validated start on login worked after restart, and the login item was removed when toggled off.

**Comment by Tas4tari** [comment]
Tested successfully on Releasee version 0.5.24 on Windows 11

**Pull request merged**